src/imap/model.rs

```diff
@@ -15,13 +16,32 @@ error_chain! {
     }
 }
 
+/// The main struct to create a connection to your imap-server.
+///
+/// # Example
+/// ```no_run
+/// use himalaya::imap::model::ImapConnector;
+/// use himalaya::config::model::Account;
+///
+/// fn main() {
+///     let account = Account::default();
+///     let mut imap_conn = ImapConnector::new(&account).unwrap();
+///
+///     // do you stuff with the connection...
+///
+///     // Be nice to the server and say 'Bye!'
+///     imap_conn.logout();
+/// }
+/// ```
 #[derive(Debug)]
 pub struct ImapConnector<'a> {
     pub account: &'a Account,
     pub sess: imap::Session<TlsStream<TcpStream>>,
 }
 
 impl<'a> ImapConnector<'a> {
+
```

Comment:
Why adding a new line here? It does not really fit the global style of the code base.